src/main/java/io/cdap/plugin/gcp/spanner/source/SpannerSource.java

```diff
@@ -241,11 +255,40 @@ private Schema getSchema(FailureCollector collector) {
                                                          projectId)) {
       DatabaseClient databaseClient =
         spanner.getDatabaseClient(DatabaseId.of(projectId, config.instance, config.database));
-      Statement getTableSchemaStatement = SCHEMA_STATEMENT_BUILDER.bind(TABLE_NAME).to(config.table).build();
+      Map<String, String> columnNameMap = new HashMap<>();
+      // get columns from import query when query does not contain the '*' or 'case'
```

Comment:
Regarding ``*`` it indicates that user selects all column of the table.
Regarding ``case`` user can type select statement like in example:
``SELECT A, B,
  CASE A
    WHEN 90 THEN 'red'
    WHEN 50 THEN 'blue'
    ELSE 'green'
  END
  AS result
FROM Numbers``
in these cases we fallback to getting full schema of the table

Reply:
What is `case` used for?

Reply:
If user types a query like: 
```
SELECT A, B, CASE A WHEN 90 THEN 'red' WHEN 50 THEN 'blue' ELSE 'green' END AS result FROM Numbers
```
in these cases we fallback to getting full schema of the table

Reply:
hm this approach is not ideal. You can construct queries where unnecessarily fallback to getting full table schema.
For example, if the query is `SELECT COUNT(*) FROM table`, we'll get the full schema. 

Instead of parsing the query, can we execute the query with the `LIMIT 1` and get the schema from the query result?

Reply:
Executing the query with `LIMIT 1` may be a better approach but it will require additional effort and time to  replace this approach and test the new approach.  So our question is whether we should continue on this PR with the new approach?

However, the new approach may not be ideal when aggregation functions are used along GROUP BY as part of the importQuery. For instance cases like:

```
SELECT COUNT(CustomerID), Country
FROM Customers
GROUP BY Country
```

Reply:
Let's implement this correctly so we won't need to rework the fix later.

Reply:
@rmstar  With the suggested implementation we face the following issue:
Currently with Spanner cloud library we can get column name and colum type. We are missing the information whether column is nullable or not.
Spanner has this information in 'ResultSetMetadata' class -however this class is onlyavailable in 'com.google.spanner' package (https://googleapis.dev/java/google-cloud-spanner/3.2.1/com/google/spanner/v1/ResultSetMetadata.html) and not in 'com.google.cloud.spanner' package ('https://googleapis.dev/java/google-cloud-spanner/latest/com/google/cloud/spanner/ResultSet.html')

Reply:
What's the effort involved to use `com.google.spanner` library to get column metadata, including whether it's nullable or not?

If that's not feasible, can we use `com.google.cloud.spanner` to get the column names, and then use the existing approach in this PR (query information_schema) to check if the column is nullable?

Reply:
We will not be able to use the alternative 1 `com.google.spanner` library.

Yes we can use the `com.google.cloud.spanner` to get the column names, and then use the existing approach (information_schema) to check if the column is nullable. However, there are cases where we can't get the nullable information like:

1. Querying using aliases
2. Aggregate Functions


For these cases should we default to nullable column?